services/speech_to_text/v1.js

```diff
@@ -74,121 +216,105 @@ module.exports = function (RED) {
       }
 
       if (!config.band) {
-        var message_err_band = 'Missing audio band configuration, unable to process speech.';
+        var message_err_band = 'Missing audio quality configuration, unable to process speech.';
 
         node.error(message_err_band, msg)
         return;
-      }
+      }  
 
+      // Has to be there, as its a checkbox, but flows switching from the old (frankly 
+      // unbeliveable) select might not have it set.
       if (!config.continuous) {
         var message_err_continuous = 'Missing continuous details, unable to process speech.';
 
         node.error(message_err_continuous, msg)
         return;
-      }
+      }  
 
-      var model = config.lang + '_' + config.band;
+      // The input comes in on msg.payload, and can either be an audio file or a string 
+      // representing a URL.  
+      if (!msg.payload instanceof Buffer || !typeof msg.payload === 'string') {
+        var message = 'Invalid property: msg.payload, can only be a URL or a Buffer.';
 
-      var speech_to_text = watson.speech_to_text({
-        username: username,
-        password: password,
-        version: 'v1',
-        url: 'https://stream.watsonplatform.net/speech-to-text/api'
-      });
+        node.error(message, msg)
+        return;
+      }
 
-      var s2t = function (audio, format, cb) {
-        if (format !== 'wav' && format !== 'flac' && format !== 'ogg') {
-          var message_err_format = 'Audio format (' + format + ') not supported, must be encoded as WAV, FLAC or OGG.';
+      // This check is repeated just before the call to the service, but 
+      // its also performed here as a double check. 
+      if (!(msg.payload instanceof Buffer)) {
+        if (typeof msg.payload === 'string' && !urlCheck(msg.payload)) {
+          var message = 'Invalid URL.';
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["message" is already defined](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2636145243/issues/source?bid=3225115&fileBranchId=3319672#l247)